code/__HELPERS/icons.dm

```diff
@@ -48,4 +85,20 @@ proc/adjust_brightness(var/color, var/value)
 	RGB[1] = Clamp(RGB[1]+value,0,255)
 	RGB[2] = Clamp(RGB[2]+value,0,255)
 	RGB[3] = Clamp(RGB[3]+value,0,255)
-	return rgb(RGB[1],RGB[2],RGB[3])
\ No newline at end of file
+	return rgb(RGB[1],RGB[2],RGB[3])
+
+/proc/AverageColor(var/icon/I)

```

Comment:
It doesn't seem to work properly. I've tested it extensively and it seems to return the blend of the bottom-rightest 2-3 pixels in the sprite, which is almost always a shade of brown due to the item having an outline. I'm working on a fixed version of this, pester me about this on the IRC